### PR TITLE
feat(installer,platform): expansion-framework support

### PR DESCRIPTION
**Pull request by MaDongdong99**
/kind feature
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/tke-installer/app/installer/expansion.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+const defaultExpansionDir = constants.DataDir + expansionDirName
+
+// expansion layout
+const expansionLayoutHooksDir = "hooks/"
+
+func (t *TKE) initExpansion(ctx context.Context) error {
+
+	if t.Config.CustomExpansionDir == "" {
+		t.Config.CustomExpansionDir = defaultExpansionDir
+		t.backup()
```
Comment (leoryu) [inline cmd/tke-installer/app/installer/expansion.go:41]: This backup func only run when t.Config.CustomExpansionDir == "". Is it right?

Comment (MaDongdong99) [reply cmd/tke-installer/app/installer/expansion.go:41]: yes, and once it is set up by default, this step will never come.

Comment (leoryu) [reply cmd/tke-installer/app/installer/expansion.go:41]: I think backup will write and update tke.json. If user input his or her CustomExpansionDir, why not write and update tke.json via backup func?

**Comment by leoryu** [comment]
LGTM

**Pull request merged**